# Ensure that the member value is stored in big-endian format when adding a member to an enum type
--
diff --git a/cxx4/ncEnumType.h b/cxx4/ncEnumType.h
index c3e9fe1..8ed32e7 100644
--- a/cxx4/ncEnumType.h
+++ b/cxx4/ncEnumType.h
@@ -70,7 +70,11 @@ namespace netCDF
       */
       template <class T> void addMember(const std::string& name, T memberValue)
       {
-	ncCheck(nc_insert_enum(groupId, myId, name.c_str(), (void*) &memberValue),__FILE__,__LINE__);
+	unsigned long long v = static_cast<unsigned long long>(memberValue);
+#if defined(__BYTE_ORDER__) && (__BYTE_ORDER__ == __ORDER_BIG_ENDIAN__)
+	v <<= (sizeof(v) - getBaseType().getSize()) * 8;
+#endif
+	ncCheck(nc_insert_enum(groupId, myId, name.c_str(), &v),__FILE__,__LINE__);
       }
 
       /*! Returns number of members in this NcEnumType object. */
